File: src/go/svd/models/electricity_consumption.go
package models

// ElectricityConsumption represents the "Electricity Consumption" section of SVD.
type ElectricityConsumption struct {
	BoilerElectricityConsumption            float64 `json:"boilerElectricityConsumption" xml:"BoilerElectricityConsumption" csv:"BoilerElectricityConsumption"`
	GeneratorProduction                     float64 `json:"generatorProduction" xml:"GeneratorProduction" csv:"GeneratorProduction"`
	OffsetElectricityConsumption            float64 `json:"offsetElectricityConsumption" xml:"OffsetElectricityConsumption" csv:"OffsetElectricityConsumption"`
	PowerConsumptionForPlant                float64 `json:"powerConsumptionForPlant" xml:"PowerConsumptionForPlant" csv:"PowerConsumptionForPlant"`
	ElectricalForCargoCooling               float64 `json:"electricalForCargoCooling" xml:"ElectricalForCargoCooling" csv:"ElectricalForCargoCooling"`
	ElectricalForDischargePump              float64 `json:"electricalForDischargePump" xml:"ElectricalForDischargePump" csv:"ElectricalForDischargePump"`
	ElectricalForReeferContainers           float64 `json:"electricalForReeferContainers" xml:"ElectricalForReeferContainers" csv:"ElectricalForReeferContainers"`
	ElectricalFromOnShorePowerSupply        float64 `json:"electricalFromOnShorePowerSupply" xml:"ElectricalFromOnShorePowerSupply" csv:"ElectricalFromOnShorePowerSupply"`
	ElectricalFromZeroEmissionsTechnologies float64 `json:"electricalFromZeroEmissionsTechnologies" xml:"ElectricalFromZeroEmissionsTechnologies" csv:"ElectricalFromZeroEmissionsTechnologies"`
	FuelTypeUsedForCargoCooling             string  `json:"fuelTypeUsedForCargoCooling" xml:"FuelTypeUsedForCargoCooling" csv:"FuelTypeUsedForCargoCooling"`
	FuelTypeUsedForDischargePump            string  `json:"fuelTypeUsedForDischargePump" xml:"FuelTypeUsedForDischargePump" csv:"FuelTypeUsedForDischargePump"`
	FuelTypeUsedForReeferContainers         string  `json:"fuelTypeUsedForReeferContainers" xml:"FuelTypeUsedForReeferContainers" csv:"FuelTypeUsedForReeferContainers"`
	FuelOilConsumptionForCargoCooling       float64 `json:"fuelOilConsumptionForCargoCooling" xml:"FuelOilConsumptionForCargoCooling" csv:"FuelOilConsumptionForCargoCooling"`
	FuelOilConsumptionForDischargePump      float64 `json:"fuelOilConsumptionForDischargePump" xml:"FuelOilConsumptionForDischargePump" csv:"FuelOilConsumptionForDischargePump"`
	FuelOilConsumptionForReeferContainers   float64 `json:"fuelOilConsumptionForReeferContainers" xml:"FuelOilConsumptionForReeferContainers" csv:"FuelOilConsumptionForReeferContainers"`
}

File: src/elixir/svd/lib/svd/models/electricity_consumption.ex
defmodule SVD.Models.ElectricityConsumption do
  @moduledoc false

  @fields [
    :total_power_consumption,
    :auxiliary_engine_consumption,
    :boiler_consumption,
    :shore_power_consumption,
    :other_consumption,
    :boiler_electricity_consumption,
    :generator_production,
    :offset_electricity_consumption,
    :power_consumption_for_plant,
    :electrical_for_cargo_cooling,
    :electrical_for_discharge_pump,
    :electrical_for_reefer_containers,
    :electrical_from_on_shore_power_supply,
    :electrical_from_zero_emissions_technologies,
    :fuel_type_used_for_cargo_cooling,
    :fuel_type_used_for_discharge_pump,
    :fuel_type_used_for_reefer_containers,
    :fuel_oil_consumption_for_cargo_cooling,
    :fuel_oil_consumption_for_discharge_pump,
    :fuel_oil_consumption_for_reefer_containers
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end
